### ISSUE TITLE
Without installing Chrome extension

### ISSUE THREAD

**Issue by johnss** [webrecorder/archiveweb.page#27]
I mostly using mobile, and nowadays more traffic come from mobile than desktop, as you already know Chrome android cannot install extension, I hope in the future creating WARC without installing anything will be possible.

I heard its not possible because to record traffic in the background it need to be a chrome extension, but i think will be possible using a virtual browser called ViewFinderJS
https://github.com/i5ik/22120 is using https://github.com/i5ik/ViewFinderJS

**Comment by ikreymer** [comment]
The browser doesn't allow archiving without either an extension, or some sort of proxy in the middle.
The Conifer approach, and the other tool you mentioned, uses such a proxy. There are tradeoffs to each approach.
Logging in to other sites will also have to go through a proxy, and they may reject logins form unknown IPs. Without an extension, all the data that'd be written to WARC is what the proxy sees, not what your device would see necessarily, which may be a big difference.

Probably what you want is an ArchiveWeb.page mobile app to do this, I think in theory it's doable, but don't really have resources to work on that right now unfortunately.

Can you say more about your use case? Do you want the data stored on your mobile phone?

**Comment by johnss** [comment]
How about localhost proxy

1. Install NodeJS using Termux
2. Install JS based proxy using npm
3. Start a proxy server
4. Tell user to use that proxy in the proxy setting
5. Proxy will log all request and write WARC in the background

**Comment by johnss** [comment]
> Can you say more about your use case?
I used it primarily to archive YouTube video, download the video and reupload to Google Drive, but now i regret it because i only archive the video file and the video title, now i want to archive the entire page because i want to archive the video description, user comments etc.
> Do you want the data stored on your mobile phone?
I will store on SD card and will upload it to Google Drive

**Comment by whalehub** [comment]
> I used it primarily to archive YouTube video, download the video and reupload to cloud storage, but now i regret it because i only archive the video file and the video title, now i want to archive the entire page because i want to archive the video description, url, comments etc.

@johnss Check out [youtube-dl](https://github.com/ytdl-org/youtube-dl); it can do all of that and more.

**Comment by johnss** [comment]
How archiving web pages possible will youtube-dl, youtube-dl do not have such functionality, maybe you mean saving video metadata in JSON file right? What i want is view video in browser exactly the same as it was before it was deleted

**Comment by whalehub** [comment]
> How archiving web pages possible will youtube-dl, youtube-dl do not have such functionality, maybe you mean saving video metadata in JSON file right? What i want is view video in browser exactly the same as it was before it was deleted

@johnss Yeah, I was talking about saving all of the metadata with youtube-dl. There are two browser on Android that can install Chrome extensions btw (Yandex Browser and Kiwi Browser); it might be worth trying it with them.

**Comment by tsemachh** [comment]
BTW: In android you can use KIWI Browser which is based on chrome , it allows to install chrome extension

**Comment by johnss** [comment]
This is electron app, I think convert to an Android app should be easy, if you want to archive using android browser you can use express.archiveweb.page
